core/collateral/engine.go

```diff
@@ -685,6 +685,50 @@ func (e *Engine) TransferRewards(ctx context.Context, rewardAccountID string, tr
 	return responses, nil
 }
 
+func (e *Engine) TransferVestedRewards(
+	ctx context.Context, transfers []*types.Transfer,
+) ([]*types.LedgerMovement, error) {
+	if len(transfers) == 0 {
+		return nil, nil
+	}
+
+	transferReqs := make([]*types.TransferRequest, 0, len(transfers))
+	for _, t := range transfers {
+		transferReqs = append(transferReqs, &types.TransferRequest{
+			FromAccount: []*types.Account{
+				e.GetOrCreatePartyVestingRewardAccount(ctx, t.Owner, t.Amount.Asset),
```

Comment:
is it possible for it not to exist?

Reply:
nah, that's why I make it panic internally, so I thought i'd be good enough.